### DISCUSSION TITLE
SagerNet with RethinkDNS

### DISCUSSION THREAD

**Discussion by ghost** [SagerNet/SagerNet#331] [Q&A]
[RethinkDNS](https://rethinkdns.com/) is a software to intercept DNS queries in order to filter them and so supports firewall functionality, it's similar to Blokada + a firewall.
RethinkDNS uses a VPN connection in order to function, also I want to use SagerNet to connect my wireguard/VPN server, I found out that in order to use both on the same device I need to use SagerNet in proxy mode and set proxy address from RethinkDNS to localhost.
I did put SagerNet in proxy mode and added it to RethinkDNS as a proxy server, it works most of times, but sometimes I'm not able to connect Internet and when I disconnect rethinkdns from proxy it works again.
Not sure if it's SagerNet issue or not, maybe someone have an advice how should I proceed?

**Comment by rurirei** [comment]
sagernet: proxy-only mode pls.

**Comment by gittyhub888** [comment]
> [RethinkDNS](https://rethinkdns.com/) is a software to intercept DNS queries in order to filter them and so supports firewall functionality, it's similar to Blokada + a firewall. RethinkDNS uses a VPN connection in order to function, also I want to use SagerNet to connect my wireguard/VPN server, I found out that in order to use both on the same device I need to use SagerNet in proxy mode and set proxy address from RethinkDNS to localhost. I did put SagerNet in proxy mode and added it to RethinkDNS as a proxy server, it works most of times, but sometimes I'm not able to connect Internet and when I disconnect rethinkdns from proxy it works again. Not sure if it's SagerNet issue or not, maybe someone have an advice how should I proceed?

Have you been able to solve this?
I tried the same configuration but failed, just like you.
Have you found a solution?
